stake-pool/single-pool/src/processor.rs

```diff
@@ -721,13 +712,8 @@ impl Processor {
             pool_mint.base.supply
         };
 
-        // XXX TODO FIXME use of lamports_added is VERY WRONG
-        // we mint tokens based on the total lamports that flow THROUGH our stake account
-        // including the rent we return!! meaning someone can steal funds
-        // we actually want to use *stake_added*. and probably (as noted at the fn def) the pre stake too
-        let new_pool_tokens =
-            calculate_deposit_amount(token_supply, pre_validator_lamports, lamports_added)
-                .ok_or(SinglePoolError::UnexpectedMathError)?;
+        let new_pool_tokens = calculate_deposit_amount(token_supply, pre_pool_stake, stake_added)
+            .ok_or(SinglePoolError::UnexpectedMathError)?;
```

Comment:
re: the above also note this nightmare bug stemming from my naively copying the lamports calculations